sdk/src/types/block/context_input/mod.rs

```diff
@@ -0,0 +1,82 @@
+// Copyright 2020-2023 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+mod reward;
+
+use derive_more::From;
+
+use self::reward::RewardContextInput;
+use crate::types::block::Error;
+
+/// A generic input supporting different input kinds.
```

Comment:
Do we have a better explanation in the TIP?

Reply:
Yes, added